src/classic/element/ReactElementValidator.js

```diff
@@ -312,8 +312,14 @@ function checkAndWarnForMutatedProps(element) {
 
   for (var propName in props) {
     if (props.hasOwnProperty(propName)) {
-      if (!originalProps.hasOwnProperty(propName) ||
-          originalProps[propName] !== props[propName]) {
+      var valueChanged = originalProps[propName] !== props[propName];
+      // Necessary because NaN !== NaN
+      if (typeof originalProps[propName] === 'number' &&
+          typeof props[propName] === 'number' &&
+          isNaN(originalProps[propName]) && isNaN(props[propName])) {
+        valueChanged = false;
+      }
+      if (!originalProps.hasOwnProperty(propName) || valueChanged) {
```

Comment:
Can you move the hasOwnProperty check before the valueChanged check? That avoids problems with possible getters on the prototype chain.